### PR TITLE
Added map page

### LINKED ISSUE #498: Create basic foundational map page to show device locations
References #497

We need a very basic map page so that we can start to create a page where users can locate devices on a map.

### PR DESCRIPTION
**Pull request by josephmoore99**
Closes #498 

This is a very basic map page that will allow users to look at a map view of devices and click to navigate to their respective pages.

Note: you will need to set your api key in your .env file in the BEIMA.Client folder.

https://user-images.githubusercontent.com/43118853/163917318-8652698e-3d16-4428-b596-b16d2d308515.mp4

### CODE REVIEW

File: BEIMA.Client/src/pages/Maps/MapPage.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                                    <div className={styles.markerContainer} onClick={() => navigate(`/devices/${device.id}`)}>
+                                      <span className={styles.markerText}>{device.deviceTag}</span>
+                                    </div>
+                                  </Marker>
+      ));
+      setLoading(false)
+      setDevices(devices)
+    }
+    loadData()
+  },[]);
```
Comment (AshlynAdamson) [inline BEIMA.Client/src/pages/Maps/MapPage.js:50]:
```suggestion
  },[navigate]);
```

**Comment by AshlynAdamson** [comment]
This looks great! My only concern is if there are a lot of devices and the markers get cluttered up, but that can be a future improvement

**Comment by josephmoore99** [comment]
> This looks great! My only concern is if there are a lot of devices and the markers get cluttered up, but that can be a future improvement

Yeah, this is super basic right now, just wanted to get the groundwork laid for next semester.

File: BEIMA.Client/src/pages/Maps/MapPage.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+  const navigate = useNavigate();
+
+  const DeviceListCall = async () => {
+    let data = await GetDeviceList();
+    return data.response;
+  }
+
+  useEffect(() => {
+    setPageName('Map')
```
Comment (AshlynAdamson) [inline BEIMA.Client/src/pages/Maps/MapPage.js:31]: We can add this to the useEffect below as the first line, make sure to also add it to the dependency array

**Review by AshlynAdamson — APPROVED** [review]

**Review by thessbsu — APPROVED** [review]

**Pull request merged**